File: main/scala/mod/iceandshadow3/basics/BlockLogic.java
package mod.iceandshadow3.basics;

import java.util.List;

import mod.iceandshadow3.compat.BMateria;
import mod.iceandshadow3.compat.CRefItem;
import mod.iceandshadow3.compat.HarvestMethod;

public class BlockLogic {
	public interface IProvider {
		public BlockLogic getBlockLogic();
	}
	protected final BMateria mat;
	public BlockLogic(BMateria m) {
		mat = m;
	}
	
	public BMateria getMateria() {return mat;}
}

File: main/scala/mod/iceandshadow3/basics/BlockLogic.scala
package mod.iceandshadow3.basics

import mod.iceandshadow3.compat.BMateria
import mod.iceandshadow3.compat.BBlockLogic
import mod.iceandshadow3.compat.CRefItem
import mod.iceandshadow3.compat.HarvestMethod

class BlockLogic(mat: BMateria) extends BBlockLogic(mat) {
	def isToolClassEffective(method: HarvestMethod): Boolean =
		this.getMateria().isToolClassEffective(method)
}
trait IBlockLogicProvider {
	def getBlockLogic(): BlockLogic
	def maxVariants(): Int
}